pkg/specgen/specgen.go

```diff
@@ -602,14 +602,17 @@ type ContainerHealthCheckConfig struct {
 	// Requires that HealthConfig be set.
 	// Optional.
 	StartupHealthConfig *define.StartupHealthCheck `json:"startupHealthConfig,omitempty"`
-	// HealthLogDestination defines the destination where the log is stored
-	HealthLogDestination string `json:"healthLogDestination,omitempty"`
+	// HealthLogDestination defines the destination where the log is stored.
+	// Nil value means the default value (local).
+	HealthLogDestination *string `json:"healthLogDestination,omitempty"`
 	// HealthMaxLogCount is maximum number of attempts in the HealthCheck log file.
-	// ('0' value means an infinite number of attempts in the log file)
-	HealthMaxLogCount uint `json:"healthMaxLogCount,omitempty"`
+	// ('0' value means an infinite number of attempts in the log file).
+	// Nil value means the default value (5).
+	HealthMaxLogCount *uint `json:"healthMaxLogCount,omitempty"`
 	// HealthMaxLogSize is the maximum length in characters of stored HealthCheck log
-	// ("0" value means an infinite log length)
-	HealthMaxLogSize uint `json:"healthMaxLogSize,omitempty"`
+	// ("0" value means an infinite log length).
+	// Nil value means the default value (500).
+	HealthMaxLogSize *uint `json:"healthMaxLogSize,omitempty"`
```

Comment:
Note you cannot change them, specgen is part of our stable API via pkg/bindings.
The pointer change I mentioned is only for the libpod container config as it doesn't have to be stable.

Reply:
Can I at least remove the `omitempty`?

Reply:
yes omitempty should be fine to remove (and I guess is needed to make `--health-max-log-count 0` work via remote, otherwise the client would not add it top the json and the server thus using the 5 as default)

Reply:
Yes, it is not serialized to `httpBody`, but parsing from `SpecGenerator.ContainerCreateCommand` is also an option. You can also add it to the payload manually. Which is the better way? @Luap99

Reply:
Okay, that simplifies the problem. I guess it depends on what priority containers.conf should have.